src/app/app.component.ts

```diff
@@ -26,81 +25,34 @@ export class AppComponent implements AfterViewInit {
   readonly typeColors = typeColors;
 
   loading = true;
-  dataSource = new MatTableDataSource<Row>(this.currentData);
+  dataSource = new MatTableDataSource<Row>([]);
 
   @ViewChild(MatTable) table!: MatTable<Row>;
   @ViewChild(MatSort) sort!: MatSort;
 
-  constructor(private readonly http: HttpClient, private readonly filterService: FilterService) {}
+  constructor(
+    private readonly filterService: FilterService,
+    private readonly appService: AppService
+  ) {}
 
   ngAfterViewInit(): void {
     this.dataSource.sort = this.sort;
 
-    this.filterService.values$.pipe(map((filters: Filters) => this.getData(filters))).subscribe({
-      next: rows$ => {
-        rows$.subscribe(rows => {
-          this.currentData = rows;
-          this.refreshTable();
-
-          this.loading = false;
-        });
-      },
-      error: e => {
-        console.error(e);
-      }
-    });
-  }
-
-  private getData(filters: Filters): Observable<Row[]> {
-    if (!this.loading) {
-      this.loading = true;
-    }
-
-    return this.http
-      .get<APIResults[]>(
-        'http://localhost:3000/' +
-          `api/characters/battleground_deserters/${filters.entriesPerPage}?from=${
-            (filters.currentPage - 1) * filters.entriesPerPage
-          }${filters.nameFilter ? '&name=' + filters.nameFilter : ''}`
-      )
+    this.filterService.values$
       .pipe(
-        map(obj => {
-          return obj.map((res, index) => {
-            const newRow: Row = {
-              ...res,
-              position: index + 1 + (filters.currentPage - 1) * filters.entriesPerPage
-            };
-
-            const datetime = new Date(newRow.datetime);
-            const formatter = Intl.DateTimeFormat(undefined, {
-              dateStyle: 'short',
-              timeStyle: 'long'
-            });
-            newRow.datetime = formatter.format(datetime);
-
-            return newRow;
-          });
-        })
-      );
-  }
-
-  private refreshTable(): void {
-    this.dataSource.data = this.currentData;
+        tap(() => {
+          this.loading = true;
+        }),
```

Comment:
I see that `tap()` is avoided when it is possible, think about a possible alternative